Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -591,6 +592,17 @@ bool NetworkStorageManager::removeOriginStorageManagerIfPossible(const WebCore::
 void NetworkStorageManager::updateLastModificationTimeForOrigin(const WebCore::ClientOrigin& origin)
 {
     assertIsCurrent(workQueue());
+
+    auto currentTime = WallTime::now();
+    auto lastModificationTime = m_lastModificationTimes.get(origin);
+    if (currentTime - lastModificationTime <= originLastModificationTimeUpdateInterval)
+        return;
+
+    m_lastModificationTimes.set(origin, currentTime);
+    m_lastModificationTimes.removeIf([&currentTime](auto& iterator) {
+        return currentTime - iterator.value > originLastModificationTimeUpdateInterval;
+    });
+
     // This function must be called when origin is in use, i.e. OriginStorageManager exists.
     auto* manager = m_originStorageManagers.get(origin);
```

Comment:
Why not storing lastModificationTime in manager instead?

Reply:
In current implementation we are actively removing OriginStorageManager when possible (see places `NetworkStorageManager::removeOriginStorageManagerIfPossible` is called), so there is a higher chance the `OriginStorageManager` goes away in the originLastModificationTimeUpdateInterval than `NetworkStorageManager`.